### PR TITLE
UNS Patching

### PR DESCRIPTION
**Pull request by lukecampbell**
Part of [OOIION-654](https://jira.oceanobservatories.org/tasks/browse/OOIION-654)

### CODE REVIEW

File: ion/services/dm/presentation/user_notification_service.py [hunk 1]
```diff
@@ -494,36 +494,34 @@ def find_events_extended(self, origin='', type='', min_time= 0, max_time=0, limi
...
-            search_origin = 'search "origin" is "*" from "events_index"'
+            query.append( 'search "origin" is "%s" from "events_index"' % origin)
 
         if type:
-            search_type = 'search "type_" is "%s" from "events_index"' % type
-        else:
-            search_type = 'search "type_" is "*" from "events_index"'
+            query.append( 'search "type_" is "%s" from "events_index"' % type)
+
+        search_string = ' and '.join(query)
```
Comment (lukecampbell) [inline ion/services/dm/presentation/user_notification_service.py:508]: The additional queries were unnecessary, these changes search ONLY for what we care about, none of the extra stuff.

File: ion/services/dm/presentation/user_notification_service.py [hunk 2]
```diff
@@ -494,36 +494,34 @@ def find_events_extended(self, origin='', type='', min_time= 0, max_time=0, limi
...
 
         log.debug("(find_events_extended) UNS found the following relevant events: %s", events)
 
-        if limit > -1:
-            list = []
-            for i in xrange(limit):
-                list.append(events[i])
-            return list
+        if limit > 0:
+            return events[:limit]
```
Comment (lukecampbell) [inline ion/services/dm/presentation/user_notification_service.py:524]: Simplification and optimization

File: ion/services/sa/test/test_activate_instrument.py [hunk 3]
```diff
@@ -433,7 +433,7 @@ def start_instrument_agent():
 
         extended_instrument = self.imsclient.get_instrument_device_extension(instrument_device_id=instDevice_id, user_id=user_id)
         log.debug( "test_activateInstrumentSample: extended_instrument %s", str(extended_instrument) )
-        log.debug( "test_activateInstrumentSample: extended_instrument computed user_notification_requests %s", str(extended_instrument.computed.user_notification_requests.value) )
+        log.debug( "test_activateInstrumentSample: extended_instrument computed user_notification_requests %s", [i.__dict__ for i in extended_instrument.computed.user_notification_requests])
```
Comment (lukecampbell) [inline ion/services/sa/test/test_activate_instrument.py:436]: Hard to see what the values are without expanding the object's attributes.

**Comment by blazetopher** [comment]
:+1:

**Pull request merged**